oxidation/libparsec/crates/core_fs/src/storage/manifest_storage.rs

```diff
@@ -0,0 +1,641 @@
+        if !cache_only {
+            self.ensure_manifest_persistent(entry_id)?;
+        }
+        Ok(())
+    }
+
+    pub fn ensure_manifest_persistent(&self, entry_id: EntryID) -> FSResult<()> {
+        match (
+            self.cache_ahead_of_localdb
+                .lock()
+                .expect("Mutex is poisoned")
+                .get(&entry_id),
+            self.cache.lock().expect("Mutex is poisoned").get(&entry_id),
+        ) {
+            (Some(pending_chunk_ids), Some(manifest)) => {
+                let ciphered = manifest.dump_and_encrypt(&self.local_symkey);
+                let pending_chunk_ids = pending_chunk_ids
+                    .iter()
+                    .map(|chunk_id| chunk_id.as_ref())
+                    .collect::<Vec<_>>();
+
+                let vlob_id = (*entry_id).as_ref();
+
+                let conn = &mut *self.conn.lock().expect("Mutex is poisoned");
+                sql_query("INSERT OR REPLACE INTO vlobs (vlob_id, blob, need_sync, base_version, remote_version)
+                    VALUES (
+                    ?, ?, ?, ?,
+                        max(
+                            ?,
+                            IFNULL((SELECT remote_version FROM vlobs WHERE vlob_id=?), 0)
+                        )
+                    )")
+                    .bind::<diesel::sql_types::Binary, _>(vlob_id)
+                    .bind::<diesel::sql_types::Binary, _>(ciphered)
+                    .bind::<diesel::sql_types::Bool, _>(manifest.need_sync())
+                    .bind::<diesel::sql_types::Integer, _>(manifest.base_version() as i32)
+                    .bind::<diesel::sql_types::Integer, _>(manifest.base_version() as i32)
+                    .bind::<diesel::sql_types::Binary, _>(vlob_id)
+                    .execute(conn)
+                    .map_err(|e| FSError::InsertTable(format!("vlobs: _ensure_manifest_persistent {e}")))?;
+
+                for pending_chunk_ids_chunk in pending_chunk_ids.chunks(SQLITE_MAX_VARIABLE_NUMBER)
+                {
+                    diesel::delete(
+                        chunks::table.filter(chunks::chunk_id.eq_any(pending_chunk_ids_chunk)),
+                    )
+                    .execute(conn)
+                    .map_err(|e| FSError::DeleteTable(format!("chunks: clear_manifest {e}")))?;
+                }
+
+                Ok(())
+            }
+            _ => Ok(()),
+        }
+    }
+
+    pub fn flush_cache_ahead_of_persistance(&self) -> FSResult<()> {
+        let keys = self
+            .cache_ahead_of_localdb
+            .lock()
+            .expect("Mutex is poisoned")
+            .keys()
+            .copied()
+            .collect::<Vec<_>>();
+        for entry_id in keys {
+            self.ensure_manifest_persistent(entry_id)?;
+        }
+
+        Ok(())
+    }
+
+    // This method is not used in the code base but it is still tested
+    // as it might come handy in a cleanup routine later
+
```

Comment:
into doc-comment

Reply:
This comment seems to be for mainteners, not for users

Reply:
ok